docs/source/developers/cpp/building.rst

```diff
@@ -124,7 +124,7 @@ argument is omitted then a release build will be produced.
    You need to more options to build on Windows. See
    :ref:`developers-cpp-windows` for details.
 
-Minimal release build:
+Minimal release build (1Gb of RAM for building or more recommended):
```

Comment:
How did you arrived at these estimates?

Reply:
There is a discussion on [JIRA](https://issues.apache.org/jira/browse/ARROW-14039), estimates arrived at by measuring when doing compilation.  It is suggested to just have an estimate of 4GB.